internal/rukpak/source/containers_image.go

```diff
@@ -41,12 +42,21 @@ func (i *ContainersImageRegistry) Unpack(ctx context.Context, bundle *BundleSour
 		return nil, reconcile.TerminalError(fmt.Errorf("error parsing bundle, bundle %s has a nil image source", bundle.Name))
 	}
 
+	srcCtx := *i.SourceContext
```

Comment:
This is a shallow copy. In this particular case, adding a `AuthFilePath` to the shallow copy won't affect anything, but it would be easy for our future selves to overlook this and accidentally modify a pointer within the shallow copy (which would affect the original)

I wonder two things:
1. If we hardcode `AuthFilePath` in `main.go` but the file doesn't exist, can `containers/image` tolerate that? I think the answer is no, but have we tried?
2. If the answer to (1) is no, maybe we need `SourceContextFunc` on the `ContainersImageRegistry` struct, which can generate a brand new `*types.SystemContext` for one-time-use when we call it (which we would do in `Unpack`. That way we can provide a function that can do the below check and optionally setup `AuthFilePath`.

Reply:
[Done](https://github.com/operator-framework/operator-controller/pull/1322/commits/33f8089a6afa7c7dffe4998e432881036f8e80c0#diff-435eecb1d2af40ee96747f88ab71eb2758da989c92f76dcb1f714fc1b300c633R230-R244)

Reply:
Nice! Looks great!